grails-test-suite-uber/src/test/groovy/grails/compiler/DomainClassWithInnerClassUsingStaticCompilationSpec.groovy

```diff
@@ -0,0 +1,60 @@
+package grails.compiler
+
+import grails.compiler.GrailsCompileStatic
+import grails.persistence.Entity
+import grails.testing.gorm.DomainUnitTest
+import grails.validation.Validateable
+import spock.lang.Issue
+import spock.lang.Specification
+
+class DomainClassWithInnerClassUsingStaticCompilationSpec extends Specification implements DomainUnitTest<SomeClass> {
+
+    @Issue('https://github.com/grails/grails-core/issues/12461')
+    void 'a domain class marked with @GrailsCompileStatic containing an inner class and a "constraints" block'() {
+        expect: 'the configuration from the "constraints" closure is available'
+            SomeClass.constraints instanceof Closure
+            SomeClass.constraintsClosureCalled
```

Comment:
Another way could be to actually define a constraint, create an entity, call `entity.validate()` and check for e.g.             `SomeClass.getConstrainedProperties().containsKey('some')` or even `entity.errors['some']` or something else. But I think this would involve more components that could break in other ways.